osquery/tables/networking/linux/tests/namespace_tests.cpp

```diff
@@ -0,0 +1,175 @@
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <gtest/gtest.h>
+
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+
+#include "osquery/tests/test_additional_util.h"
+#include "osquery/tests/test_util.h"
+
+#include <netinet/in.h>
+#include <sched.h>
+#include <signal.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+namespace {
+const int kNamespaceTestPort = 27960;
+
+// It is easier to allocate these structures here instead of inside the
+// new entry point (which may cause stack issues)
+struct sockaddr_in server_addr = {};
+struct sockaddr_in client_addr = {};
+
+// The new entry point used by the listening_ports_table test; this
+// code just listens for a new client and terminates after a connection
+// has been received
+int NetNamespaceTests_CloneEntryPoint(void*) {
+  server_addr.sin_family = AF_INET;
+  server_addr.sin_addr.s_addr = htonl(INADDR_ANY);
+  server_addr.sin_port = htons(kNamespaceTestPort);
+
+  auto server_socket = socket(AF_INET, SOCK_STREAM, 0);
+  if (server_socket == -1) {
+    std::cerr << "Failed to create the socket\n";
+    return 1;
+  }
+
+  if (bind(server_socket,
+           reinterpret_cast<struct sockaddr*>(&server_addr),
+           sizeof(server_addr)) != 0) {
+    std::cerr << "bind() has failed\n";
+    return 1;
+  }
+
+  if (listen(server_socket, 1) != 0) {
+    std::cerr << "listen() has failed\n";
+    return 1;
+  }
+
+  socklen_t client_addr_size = sizeof(client_addr);
+  auto client_socket = accept(server_socket,
+                              reinterpret_cast<struct sockaddr*>(&client_addr),
+                              &client_addr_size);
+  if (client_socket == -1) {
+    std::cerr << "accept() has failed\n";
+    return 1;
+  }
+
+  close(client_socket);
+  close(server_socket);
+
+  return 0;
+}
+} // namespace
+
+namespace osquery {
+namespace tables {
+class NetNamespaceTests : public testing::Test {};
+
+TEST_F(NetNamespaceTests, listening_ports_table) {
+  // Get the current net namespace
+  std::string net_path = "/proc/" + std::to_string(getpid()) + "/ns/net";
+
+  char current_net_namespace[PATH_MAX] = {};
+  if (readlink(net_path.data(), current_net_namespace, PATH_MAX - 1) == -1) {
+    std::cerr << "Failed to acquire the current net namespace. Are namespaces "
+                 "enabled? Skipping this test...";
+    return;
+  }
+
+  // Clone the process while also creating a new net namespace
+  std::uint8_t new_stack[4096U];
+  auto new_stack_ptr = new_stack + sizeof(new_stack) - 1U;
+
+  auto child_id =
+      clone(NetNamespaceTests_CloneEntryPoint, new_stack_ptr, CLONE_NEWNET, 0);
+  EXPECT_NE(child_id, -1);
+  if (child_id == -1) {
+    std::cerr << "clone() has failed; root privileges are required to create a "
+                 "new namespace\n";
+    return;
+  }
+
+  sleep(1);
```

Comment:
can this be a `pauseMilli` to pause the thread as opposed to `sleep`?

Reply:
Sure!